### DISCUSSION TITLE
Anyone managed to detect when browser tabs are closed and respond to the closing of said tab?

### DISCUSSION THREAD

**Discussion by megalithic** [Hammerspoon/hammerspoon#3663] [Q&A] [answered]
Basically, I'm locking to automate toggling of things if a google meet tab is closed in Brave browser. Any suggestions for things that have worked for you?

**Comment by muescha** [comment]
at first you can check if the tab is already open. other idea: I would try a javascript inside the browser listening to tab closing (maye with `beforeunload` event) and save this somehow. then ask with hammerspoon via javascript for applications for this stored value.

**Reply by megalithic** [reply, marked as answer]
Here's what I ended up with.. basically watching tab count and the domain existing or not.. hacky, but _works_.

https://github.com/megalithic/dotfiles/blob/e7179ae395c10514e6106687e040b6631c22480f/config/hs/contexts/com.brave.Browser.nightly.lua?plain=1#L12-L29

**Reply by muescha** [reply]
this line?
https://github.com/megalithic/dotfiles/blob/main/config/hammerspoon/watchers/url.lua#L59

**Reply by megalithic** [reply]
Sorry, yes, I should have closed this as I’ve since then got things working.
